### PR TITLE
chore(weave): LiteLLM Support + Stream-by-flag

### PR DESCRIPTION
**Pull request by tssweeney**
We need to support a common vendor pattern: streaming is a flag, not a different symbol. This PR adds support for this case as well as support & tests for LiteLLM. Specifically:

1. Extends the Op `on_output` handler to accept inputs as well to handle more advanced cases
2. Extends the accumulator pattern to dynamically decide if accumulation is appropriate and to have a post-processor
3. Implements the LiteLLM integration + unit tests for the integration.
4. (unrelated, but helpful) Redacts `api_key` from inputs by default

Screenshot 2024-05-07 at 15 40 56

### CODE REVIEW

**Comment by krrishdholakia** [comment]
Thanks for integrating litellm here @tssweeney curious - do you also use the proxy internally?

**Comment by tssweeney** [comment]
> Thanks for integrating litellm here @tssweeney curious - do you also use the proxy internally?

Hey @krrishdholakia! We do not use the LiteLLM proxy internally, but would be interested in learning more about the capabilities. I'm still working on this PR, but soon i'll have some screenshots to show the results of the integration

**Comment by krrishdholakia** [comment]
Hey @tssweeney here's the quick start - https://docs.litellm.ai/docs/proxy/quick_start

File: weave/integrations/litellm/litellm.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+def make_wrapper(name: str) -> typing.Callable:
+    def litellm_wrapper(fn: typing.Callable) -> typing.Callable:
+        op = weave.op()(fn)
+        op.name = name  # type: ignore
+        return add_accumulator(
+            op,  # type: ignore
+            litellm_accumulator,
+            should_accumulate=should_use_accumulator,
+            on_finish_post_processor=litellm_on_finish_post_processor,
```
Comment (tssweeney) [inline weave/integrations/litellm/litellm.py:95]: These two lines are the big "feature add" for integrators.

**Review by shawnlewis — APPROVED** [review]

**Pull request merged**